ServerCore/Pages/Teams/Details.cshtml.cs

```diff
@@ -163,13 +163,58 @@ public async Task<IActionResult> OnGetAddMemberAsync(int teamId, int userId, int
             {
                 return Forbid();
             }
-            
+
             Tuple<bool, string> result = TeamHelper.AddMemberAsync(_context, Event, EventRole, teamId, userId).Result;
             if (result.Item1)
             {
                 return RedirectToPage("./Details", new { teamId = teamId });
             }
             return NotFound(result.Item2);
         }
+
+        public async Task<IActionResult> OnGetRejectMemberAsync(int teamId, int userId, int applicationId)
+        {
+            var authResult = await AuthChecks(teamId);
+            if (!authResult.passed)
+            {
+                return authResult.redirect;
+            }
+
+            if (EventRole == EventRole.play && !Event.IsTeamMembershipChangeActive)
+            {
+                return NotFound("Team membership change is not currently active.");
+            }
+
+            TeamApplication application = await (from app in _context.TeamApplications
+                                                 where app.ID == applicationId
+                                                 select app).FirstOrDefaultAsync();
+            if (application == null)
+            {
+                return NotFound("Could not find application");
+            }
+
+            if (application.Player.ID != userId)
+            {
+                return NotFound("Mismatched player and application");
```

Comment:
Is there a reason we're passing the player around in addition to the application? It seems like we might be able to just go off of that and simplify this a bit